src/main/java/com/tester/processor/BaseProcessor.java

```diff
@@ -0,0 +1,296 @@
+package com.tester.processor;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.jacoco.core.analysis.IClassCoverage;
+import org.jacoco.core.analysis.IMethodCoverage;
+
+
+public abstract class BaseProcessor {
+
+  private static final String CODE_START = "```java\n";
+  private static final String CODE_END = "```";
+  private static final String REGRESSION = "regression";
+  private static final String WORKING = "working";
+  private static final String SRC = "src/";
+  private static final String PACKAGE = "package ";
+  private static final String MAIN = "main/java/";
+  private static final String TEST = "test/java/";
+  private static final String CLASS = ".class";
+  private static final String EXAMPLES = "examples";
+  private static final String TARGET_DIR = "target/classes/";
+  private final String programFileName;
+  private final String testFileName;
+  private final String targetMethod;
+  private final Map<String, List<Integer>> suspiciousLines;
+
+  public BaseProcessor(String programFileName,
+                       String testFileName,
+                       String targetMethod,
+                       Map<String, List<Integer>> suspiciousLines) {
+    this.programFileName = programFileName;
+    this.testFileName = testFileName;
+    this.targetMethod = targetMethod;
+    this.suspiciousLines = suspiciousLines;
+  }
+
+  public static String getMAIN() {
+    return MAIN;
```

Comment:
Could you remove all these getter methods and make the fields `protected` instead? So the child classes can access those fields without having to call these getter methods.
For example 
```java
protected static final String MAIN = "main/java/";
```